### PR TITLE
✨(project) add backend boilerplate

### PR DESCRIPTION
**Pull request by wilbrdt**
## Purpose

A base FastAPI project that can trigger Celery tasks.
For now we may not need a database, it can be added later.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+              only: /.*/
+
+      # Backend jobs
+      #
+      # Build, lint and test production and development Docker images
+      # (debian-based)
+      - build:
+          matrix:
+            parameters:
+              python-image: [python:3.8, python:3.9, python:3.10, python:3.11, python:3.12]
```
Comment (lunika) [inline .circleci/config.yml:419]: Do we really want to test against all of this version ? IMO it is useless, the production image will be build with python 3.12, we can have all the benefit and feature available with python 3.12

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+
+# -- Core --
+FROM base AS core
+
+RUN apt-get update && \
+    apt-get install -y \
+        curl \
+        jq \
+        wget && \
```
Comment (lunika) [inline Dockerfile:31]: Do we really need them ?

File: README.md [hunk 3]
```diff
@@ -12,6 +41,19 @@ We try to raise our code quality standards and expect contributors to follow
...
+
+- `make lint` to run the linters
+- `make test` to run the tests
+
+Note that we also provide a git pre-commit hook to ease your life:
+```
+make git-hook-pre-commit
+```
+
+>>>>>>> 1e5cb8e (🎉(project) initialize the project)
```
Comment (lunika) [inline README.md:56]: should be removed

File: bin/git-hook-pre-commit [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+git diff-index --check --cached "$against" --
+
+# Run the necessary commands to check modified code sources.
+if git diff --staged --cached --name-only | grep -q 'src/api'; then
+  make lint-api
+fi
+if git diff --staged --cached --name-only | grep -q 'src/app'; then
+  make lint-app
+fi
+if git diff --staged --cached --name-only | grep -q 'src/frontend'; then
```
Comment (lunika) [inline bin/git-hook-pre-commit:33]: there is no front for now and I don't know if one will be added

File: bin/git-hook-pre-commit [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+# Check for trailing whitespace errors.
+echo "pre-commit hook: checking trailing whitespace…"
+git diff-index --check --cached "$against" --
+
+# Run the necessary commands to check modified code sources.
+if git diff --staged --cached --name-only | grep -q 'src/api'; then
+  make lint-api
+fi
+if git diff --staged --cached --name-only | grep -q 'src/app'; then
```
Comment (lunika) [inline bin/git-hook-pre-commit:30]: `src/app` does not exists

File: bin/pytest [hunk 6]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+declare DOCKER_USER
+DOCKER_USER="$(id -u):$(id -g)"
+
+DOCKER_USER=${DOCKER_USER} docker compose run --rm api pytest "$@"
```
Comment (lunika) [inline bin/pytest:6]: why not using `bin/compose` ?

File: Dockerfile (hunk 2 above)
Comment (wilbrdt) [reply Dockerfile:31]: Nope!

File: bin/git-hook-pre-commit (hunk 4 above)
Comment (wilbrdt) [reply bin/git-hook-pre-commit:33]: Not planned for the moment, removing it!

File: bin/pytest (hunk 6 above)
Comment (wilbrdt) [reply bin/pytest:6]: It seems we can indeed, thanks!

**Review by lunika — APPROVED** [review]

**Pull request merged**